### ISSUE TITLE
[prefer-function-type] doc comment deleted when fixed

### ISSUE THREAD

**Issue by tao-cumplido** [typescript-eslint/typescript-eslint#3004]
- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [read the FAQ](https://github.com/typescript-eslint/typescript-eslint/blob/master/docs/getting-started/linting/FAQ.md) and my problem is not listed.

**Repro**

```JSON
{
  "rules": {
    "@typescript-eslint/prefer-function-type": ["error"]
  }
}
```

```TS
interface X {
  /** @deprecated */
  (): void;
}
```

**Expected Result**

```ts
type X =
  /** @deprecated */
  () => void;
```

**Actual Result**

```ts
type X = () => void;
```

**Additional Info**

Maybe it'd be nice to also have an option to prevent transform when a doc comment is present.
I didn't check if regular comments get eaten too.

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `4.14.2` |
| `@typescript-eslint/parser`        | `4.14.2` |
| `TypeScript`                       | `4.1.3` |
| `ESLint`                           | `7.19.0` |
| `node`                             | `14.15.4` |

**Comment by armano2** [comment]
> Maybe it'd be nice to also have an option to prevent transform when a doc comment is present.

yes they are, we should disable fixer when comments are detected

**Comment by arminyahya** [comment]
Does this change include all "rules"?

**Comment by armano2** [comment]
no, this should be added only to this use case from this rule

**Comment by arminyahya** [comment]
I'm going to do this, give me one day.
